go/mysql/binlog_event.go

```diff
@@ -124,9 +124,9 @@ type BinlogEvent interface {
 	// IsWriteRows(), IsUpdateRows(), or IsDeleteRows() returns
 	// true.
 	Rows(BinlogFormat, *TableMap) (Rows, error)
-	// TransactionPayload returns a list of BinlogEvents contained
+	// TransactionPayload returns a function used to iterate over the events
 	// within the compressed transaction.
-	TransactionPayload(BinlogFormat) ([]BinlogEvent, error)
+	TransactionPayload(BinlogFormat) (func() (BinlogEvent, error), error)
```

Comment:
Looks like you're changing this to be compatible with the new Go v1.23 iterators, right? 💯

Reply:
Yeah, we will almost certainly be using 1.23 on main before the release and I'd like to use the new standard interface if I can.